calculator.c

```diff
@@ -0,0 +1,114 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+#define SIZE 100
+
+int evaluation(char operator, int op1, int op2){
+    switch(operator){
+        case '+':
+            return op1 + op2; 
+        case '-': 
+            return op1 - op2;
+        case '*': 
+            return op1 * op2;
+        case '/':
+            if(op2== 0){ 
+                printf("Error: Division by zero.");
+                exit(EXIT_FAILURE);
+            }
+            else{
+                return op1 / op2; 
+            }
+        default:
+            printf("Error: Invalid operator.");
+            exit(EXIT_FAILURE);
+    }
+}
+
+
+int precedence(char operator){
+    if(operator == '+' || operator == '-'){
+        return 1;
+    }
+    if(operator == '*' || operator == '/'){
+        return 2;
+    } 
+    return 0;
+}
+
+
+int EvaluateExpression(char *expr){

```

Comment:
This function can be divided into small functions which do one specific task. Practice writing functions in such a way that one function does one task.